src/core/thread/mesh_forwarder.cpp

```diff
@@ -1096,6 +1096,17 @@ void MeshForwarder::HandleReceivedFrame(Mac::RxFrame &aFrame)
             LogFrame("Received empty payload frame", aFrame, OT_ERROR_NONE);
         }
 
+#if OPENTHREAD_THREAD_VERSION >= OPENTHREAD_THREAD_VERSION_1_2 && OPENTHREAD_FTD
+        // From Thread 1.2, MAC Data Frame can also act as keep-alive message if child supports
+        {
```

Comment:
Do we need to check if the received frame is secure (has link security)?

Reply:
Yes, I think we sure make sure MAC frame security is enabled.

Reply:
Agree!